lib/backend/k8s/resources/systemnetworkpolicies.go

```diff
@@ -0,0 +1,205 @@
+		Resource(SystemNetworkPolicyResourceName).
+		Namespace("kube-system").
+		Body(tpr)
+	err := req.Do().Into(&res)
+	if err != nil {
+		return nil, K8sErrorToCalico(err, kvp.Key)
+	}
+	kvp.Revision = res.Metadata.ResourceVersion
+	return kvp, nil
+}
+
+// Update implements the Update method for System Network Policies (exposed
+// through the libcalico-go API as standard Policy resources)
+func (c *snpClient) Update(kvp *model.KVPair) (*model.KVPair, error) {
+	tpr := SystemNetworkPolicyToThirdParty(kvp)
+	res := thirdparty.SystemNetworkPolicy{}
+	req := c.tprClient.Put().
+		Resource(SystemNetworkPolicyResourceName).
+		Namespace("kube-system").
+		Body(tpr).
+		Name(tpr.Metadata.Name)
+	err := req.Do().Into(&res)
+	if err != nil {
+		return nil, K8sErrorToCalico(err, kvp.Key)
+	}
+	kvp.Revision = tpr.Metadata.ResourceVersion
+	return kvp, nil
+}
+
+// Apply implements the Apply method for System Network Policies (exposed
+// through the libcalico-go API as standard Policy resources)
+func (c *snpClient) Apply(kvp *model.KVPair) (*model.KVPair, error) {
+	updated, err := c.Update(kvp)
+	if err != nil {
+		if _, ok := err.(errors.ErrorResourceDoesNotExist); !ok {
+			return nil, err
+		}
+
+		// It doesn't exist - create it.
+		updated, err = c.Create(kvp)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return updated, nil
+}
+
+// Delete implements the Delete method for System Network Policies (exposed
+// through the libcalico-go API as standard Policy resources)
+func (c *snpClient) Delete(kvp *model.KVPair) error {
+	result := c.tprClient.Delete().
+		Resource(SystemNetworkPolicyResourceName).
+		Namespace("kube-system").
+		Name(systemNetworkPolicyTprName(kvp.Key)).
+		Do()
+	return K8sErrorToCalico(result.Error(), kvp.Key)
+}
+
+// Get implements the Get method for System Network Policies (exposed
+// through the libcalico-go API as standard Policy resources)
+func (c *snpClient) Get(key model.Key) (*model.KVPair, error) {
+	tpr := thirdparty.SystemNetworkPolicy{}
+	err := c.tprClient.Get().
+		Resource(SystemNetworkPolicyResourceName).
+		Namespace("kube-system").
+		Name(systemNetworkPolicyTprName(key)).
+		Do().Into(&tpr)
+	if err != nil {
+		return nil, K8sErrorToCalico(err, key)
+	}
+
+	return ThirdPartyToSystemNetworkPolicy(&tpr), nil
+}
+
+// List implements the List method for System Network Policies (exposed
+// through the libcalico-go API as standard Policy resources)
+func (c *snpClient) List(list model.ListInterface) ([]*model.KVPair, error) {
+	kvps := []*model.KVPair{}
+	l := list.(model.PolicyListOptions)
+
+	// If the Name is specified, k8s will return a single resource
+	// rather than a list, so handle this case separately, using our
+	// Get method to return the single result.
+	if l.Name != "" {
+		log.Info("Performing System Network Policy List with name")
+		if kvp, err := c.Get(model.PolicyKey{Name: l.Name}); err == nil {
+			kvps = append(kvps, kvp)
+		} else {
+			if !kerrors.IsNotFound(err) {
```

Comment:
I think the errors returned from `Get` are Calico errors, not Kubernetes errors.